### PR TITLE
[MLMD Lineage] Improve target card selection

### PR DESCRIPTION
**Pull request by kwasi**
/assign @avdaredevil 
/area front-end

## Description

This PR addresses the 3 sub-tasks of https://github.com/kubeflow/frontend/issues/16

* The selected target is now always displayed as a filled in blue dot
* The user can click anywhere on the card to target an artifact. This is shown with a cursor and a radio button hint.
* The radio button is hidden when the user's click target is the card title which navigates on click rather than selecting the target card. 

https://screencast.googleplex.com/cast/NDU2Njc4ODc0MDIxODg4MHw3Y2QxZjliOS0zNg

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/frontend/28)

### CODE REVIEW

File: src/mlmd/LineageCardRow.tsx [hunk 1]
```diff
@@ -166,14 +202,80 @@ export class LineageCardRow extends React.Component<LineageCardRowProps> {
...
+    );
   }
 
   private handleClick() {
     if (!this.props.setLineageViewTarget || !(this.props.type === 'artifact')) return;
     this.props.setLineageViewTarget(this.props.resource as Artifact);
   }
+
+  private handleMouseEnter = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    if (!e || !e.target) {
```
Comment (avdaredevil) [inline src/mlmd/LineageCardRow.tsx]:
```suggestion
    if (!e || !e.target) return;
```

File: src/mlmd/LineageCardRow.tsx [hunk 2]
```diff
@@ -166,14 +202,80 @@ export class LineageCardRow extends React.Component<LineageCardRowProps> {
...
+        <input
+          type='radio'
+          className='form-radio'
+          name=''
+          value=''
+          onClick={this.handleClick}
+          checked={this.props.isTarget}
+          readOnly={true}
+        />
+        <input
```
Comment (avdaredevil) [inline src/mlmd/LineageCardRow.tsx:220]: Why are there 2 radios?

Comment (kwasi) [reply src/mlmd/LineageCardRow.tsx:220]: One is used to provide a selection hint when the card is hovered.

The natural hover state of the material radio button does not show a hint so adding a `:hover` isn't very obvious what hovering is actually doing. Now that there's always a selected target card, there's a grey hint when you hover the card that goes away when you hover the link to indicate targeting vs following the hyperlink.

See the effect in the screencast.

File: src/mlmd/LineageCardRow.tsx (hunk 1 above)
Comment (kwasi) [reply src/mlmd/LineageCardRow.tsx]: Done.

**Comment by kwasi** [comment]
/hold cancel

Merged hotfix.

Deployed in Metadata at: https://kubeflow-ui-07.endpoints.kwasinti-kubeflow-dev.cloud.goog/.

File: src/mlmd/LineageCardRow.tsx (hunk 2 above)
Comment (avdaredevil) [reply src/mlmd/LineageCardRow.tsx:220]: Looks fantastic! Thanks :)

**Pull request merged**